app/jobs/get_suggestions_job.rb

```diff
@@ -0,0 +1,93 @@
+class GetSuggestionsJob < ApplicationJob
+  queue_as :default
+
+  def perform(job_status)
+    @job_id = job_status.job_id
+
+    send_spotify_request
+
+    if get_spotify_tracks
+      job_status.update(job_attributes_success)
+    else
+      job_status.update(job_attributes_failure)
+    end
+    job_status.done = true
+    job_status.save!
+  end
+
+  private
+
+  def client_token
+    Base64.strict_encode64(
+      "#{ENV['SPOTIFY_CLIENT_ID']}:#{ENV['SPOTIFY_CLIENT_SECRET']}",
+    ).delete("\n")
+  end
+
+  def send_spotify_request
+    token_url = "https://accounts.spotify.com/api/token"
+    grant_type = { grant_type: "client_credentials" }
+    authorization_basic = { "Authorization": "Basic #{client_token}" }
+
+    @spotify_token = JSON.parse(
+      RestClient.post(token_url, grant_type, authorization_basic),
+    )
+  end
+
+  def get_spotify_tracks
+    track_id = random_track
+    track_url = "https://api.spotify.com/v1/tracks/#{track_id}"
+    access_token = @spotify_token.values[0]
+    authorization_bearer = { "Authorization": "Bearer #{access_token}" }
```

Comment:
There's a lot of functionality in this class that doesn't necessarily relate to the background job itself. What would you say to breaking the spotify stuff out into its own class?